File: frontend/lib/models/task.dart
class Task {
  final String? id;
  final String taskName;
  final String date;
  final bool isCompleted;

  Task({
    this.id,
    required this.taskName,
    required this.date,
    this.isCompleted = false,
  });

  factory Task.fromJson(Map<String, dynamic> json) {
    return Task(
      id: json['_id'],
      taskName: json['taskName'],
      date: json['date'],
      isCompleted: json['isCompleted'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'taskName': taskName,
      'date': date,
      'isCompleted': isCompleted,
    };
  }
}

File: backend/src/models/Task.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface ITask extends Document {
  userId: mongoose.Types.ObjectId;
  taskName: string;
  date: Date;
  isCompleted: boolean;
}

const TaskSchema: Schema = new Schema({
  userId: { type: Schema.Types.ObjectId, ref: 'User', required: true },
  taskName: { type: String, required: true },
  date: { type: Date, required: true },
  isCompleted: { type: Boolean, default: false },
}, { timestamps: true });

TaskSchema.index({ userId: 1, date: 1 });

export default mongoose.model<ITask>('Task', TaskSchema);
